### PR TITLE
Issue #SB-0000 feat: Functional tests for Accept Flag Api

### PR DESCRIPTION
**Pull request by krgauraw**

### CODE REVIEW

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+import static com.jayway.restassured.http.ContentType.JSON;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+
+public class ContentAcceptFlagV3Test extends BaseTest {
+
+    private static final String BASE_PATH = "/content/v3";
+    private static final String DEFAULT_CONTENT_IMAGE_OBJECT_SUFFIX = ".img";
+
+    private static ClassLoader classLoader = ContentRetireV3Test.class.getClassLoader();
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: Please Correct the Class Name.

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    private void flag(String contentId) {
+        String versionKey = getVersionKey(contentId);
+        String flagRequestBody = "{\"request\": {\"flagReasons\":[\"Copyright Violation\"],\"flaggedBy\":\"gauraw\",\"versionKey\": \"" + versionKey + "\"}}";
+        given().
+                spec(getRequestSpecification(contentType, userId, APIToken)).
+                body(flagRequestBody).
+                with().
+                contentType(JSON).
+                when().
+                post("/content/v3/flag/" + contentId).
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: BASE_PATH should be used

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+                then().
+                body("responseCode", equalTo("OK"));
+    }
+
+    private void upload(String contentId) {
+        given().
+                spec(getRequestSpecification(uploadContentType, userId, APIToken)).
+                multiPart(new File(filePath + "/pdf.pdf")).
+                when().
+                post("/content/v3/upload/" + contentId).
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: BASE_PATH should be used

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+        String contentId = createDocumentContent();
+        upload(contentId);
+        publish(contentId);
+        delay(25000);
+        flag(contentId);
+        acceptFlag(contentId);
+        Response imageNodeResponse = getContentResponse(contentId + DEFAULT_CONTENT_IMAGE_OBJECT_SUFFIX);
+        assertEquals("FlagDraft", imageNodeResponse.jsonPath().get("result.content.status"));
+        Response response = getContentResponse(contentId);
+        assertEquals("Retired", response.jsonPath().get("result.content.status"));
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: assert statement is missing for Response Code.

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+    }
+
+    @Test
+    public void acceptFlagForDocumentContentWithDraftStatus() {
+        setURI();
+        String contentId = createDocumentContent();
+        upload(contentId);
+        Response response = acceptFlagAndGetResponse(contentId);
+        assertEquals("CLIENT_ERROR", response.jsonPath().get("responseCode"));
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: please put assert for exact error code also in all applicable test cases

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+                with().
+                    contentType(JSON).
+                when().
+                    get(BASE_PATH + "/read/" + contentId).
+                then().
+                    body("responseCode", equalTo("OK")).
+                extract().
+                response();
+    }
+
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: Please Provide Test Description for all test cases as comment in below format:
Given, When, Then

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+public class ContentAcceptFlagV3Test extends BaseTest {
+
+    private static final String BASE_PATH = "/content/v3";
+    private static final String DEFAULT_CONTENT_IMAGE_OBJECT_SUFFIX = ".img";
+
+    private static ClassLoader classLoader = ContentRetireV3Test.class.getClassLoader();
+    private static File filePath = new File(classLoader.getResource("UploadFiles/").getFile());
+
+    private final String createDocumentContentRequestBody = "{\"request\": {\"content\": {\"name\": \"Text Book 1\",\"code\": \"test.book.1\",\"mimeType\": \"application/pdf\",\"contentType\":\"Resource\"}}}";
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: Please use a fixed prefix for Identifier. e.g: "identifier":"FT_"+<Random Number generated at run time >

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+        assertEquals("FlagDraft", imageNodeResponse.jsonPath().get("result.content.status"));
+        Response response = getContentResponse(collectionContentId);
+        assertEquals("Retired", response.jsonPath().get("result.content.status"));
+    }
+
+    @Test
+    public void acceptFlagForCollectionContentWithDraftStatus() {
+        setURI();
+        String collectionContentId = createCollectionContent();
+        hierarchyUpdate(collectionContentId);
```
Comment (krgauraw) [inline platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: Hierarchy Update is not required for this scenario.

File: platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java (hunk 4 above)
Comment (shubham-v) [reply platform-tests/functional-tests/src/test/java/org/ekstep/platform/content/ContentAcceptFlagV3Test.java]: Validation for response code is written in getContentResponse(String) method.

**Pull request merged**